python/ml4ir/applications/classification/pipeline.py

```diff
@@ -160,6 +160,16 @@ def get_relevance_dataset(
 
         return relevance_dataset
 
+    def create_pipeline_for_kfold(self, args):
+        """
+        Create a ClassificationPipeline object used in running kfold cross validation.
+        """
+        return ClassificationPipeline(args=args)
```

Comment:
Why is this needed if all we are doing is calling `__init__`?

Reply:
because we can be creating either a classification pipeline or a ranking pipeline

Reply:
But aren't you creating an instance of the respective class in this method? Why would you create ranking pipeline in ClassificationPipeline class?